Modify rule S6781: Adjustments for Python (APPSEC-1693) (#3728)
SonarSource/rspec: rules/S6781/python/how-to-fix-it/django.adoc

== How to fix it in Django

> The following noncompliant Django REST framework application contains a hard-coded secret that can be exposed unintentionnaly. A solution is to set this secret in an environment variable.

Reviewer: ```suggestion
=== Code examples

==== Noncompliant code example

The following noncompliant Django REST framework application contains a hard-coded secret that can be exposed unintentionally.
```

[source,python,diff-id=401,diff-type=noncompliant]
----
INSTALLED_APPS = [
    'django.contrib.admin',
    'django.contrib.auth',
    'django.contrib.contenttypes',
    'django.contrib.sessions',
    'django.contrib.messages',
    'django.contrib.staticfiles',
    'rest_framework',
]

REST_FRAMEWORK = {
    'DEFAULT_PERMISSION_CLASSES': [
        'rest_framework.permissions.DjangoModelPermissionsOrAnonReadOnly',
    ],
    'DEFAULT_AUTHENTICATION_CLASSES': (
      'rest_framework_simplejwt.authentication.JWTAuthentication',
    )
}

SIMPLE_JWT = {
  "ALGORITHM": "HS256",
  "SIGNING_KEY": "secret"  # Noncompliant
}
----

> 

Reviewer: ```suggestion

==== Compliant solution

A solution is to set this secret in an environment variable.

```
